native/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,181 @@
+
+  const initializeTts = useCallback((): void => {
+    Tts.getInitStatus().catch(async error => {
+      reportError(`Tts-Error: ${error.code}`)
+      if (error.code === 'no_engine') {
+        await Tts.requestInstallEngine().catch((e: string) => reportError(`Failed to install tts engine: : ${e}`))
+      }
+    })
+  }, [])
+
+  const play = useCallback(
+    (index = sentenceIndex) => {
+      Tts.stop()
+      const sentence = sentences[index]
+      if (sentence) {
+        setIsPlaying(true)
+        Tts.speak(sentence, {
+          androidParams: {
+            KEY_PARAM_PAN: 0,
+            KEY_PARAM_VOLUME: 0.6,
+            KEY_PARAM_STREAM: 'STREAM_MUSIC',
+          },
+          iosVoiceId: '',
+          rate: 1,
+        })
+      }
+    },
+    [sentenceIndex, sentences],
+  )
+
+  const stop = async () => {
+    setIsPlaying(false)
+    setSentenceIndex(0)
+    await Tts.stop()
+    const TTS_STOP_DELAY = 100
+    await new Promise(resolve => {
+      setTimeout(resolve, TTS_STOP_DELAY)
+    })
+  }
+
+  const pause = () => {
+    Tts.stop()
+    setIsPlaying(false)
+  }
+
+  const playNext = useCallback(
+    (index = sentenceIndex) => {
+      const nextIndex = index + 1
+      if (nextIndex < sentences.length) {
+        setSentenceIndex(nextIndex)
+        play(nextIndex)
+      } else {
+        stop()
+      }
+    },
+    [play, sentenceIndex, sentences.length],
+  )
+
+  const playPrevious = (index = sentenceIndex) => {
+    const prevIndex = Math.max(0, index - 1)
+    if (prevIndex < sentences.length) {
+      setSentenceIndex(prevIndex)
+      play(prevIndex)
+    }
+  }
+
+  useEffect(() => {
+    if (!enabled) {
+      return () => undefined
+    }
+
+    initializeTts()
+    Tts.addEventListener('tts-finish', () => playNext(sentenceIndex))
+
+    return () => {
+      Tts.removeAllListeners('tts-finish')
+    }
+  }, [enabled, initializeTts, playNext, sentenceIndex, sentences.length])
+
+  useEffect(() => {
+    if (sentences.length <= 1) {
+      stop()
+    }
+  }, [sentences])
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      if (nextAppState === 'inactive' || nextAppState === 'background') {
+        stop()
+      }
+    })
+
+    return subscription.remove
+  }, [])
+
+  const close = async () => {
+    setVisible(false)
+    await stop()
+  }
+
+  const ttsContextValue = useMemo(
+    () => ({
+      enabled,
+      setEnabled,
+      sentenceIndex,
+      setSentenceIndex,
+      visible,
+      setVisible,
+      sentences,
+      setSentences,
+      languageCode,
+    }),
+    [enabled, sentenceIndex, visible, sentences, languageCode],
+  )
+
+  return (
+    <ttsContext.Provider value={ttsContextValue}>
+      {children}
+      {visible && (
+        <TtsPlayer
+          isPlaying={isPlaying}
+          setIsPlaying={setIsPlaying}
+          sentenceIndex={sentenceIndex}
```

Comment:
:wrench: Both `setIsPlaying` and `sentenceIndex` should not be necessary anymore, could you adjust `TtsPlayer` accordingly?

Reply:
Passing `sentenceIndex` is needed for `playPrevious(sentenceIndex)` and `playNext(sentenceIndex)`

Reply:
There is no need to pass sentence index to these functions, they can just directly read and use it

Reply:
If I pass it directly it won't keep playing.. it will stop at an index for some reason.